backend/api/v2beta1/artifacts.proto

```diff
@@ -0,0 +1,125 @@
+  // Unique Artifact ID. Generated by MLMD.
+  string artifact_id = 1;
+  // Storage Provider to which this Artifact is located (e.g. S3, Minio, etc.).
+  string storage_provider = 3;
+  // The path location of this Artifact within the storage provider.
+  // For example an object located at s3://my-bucket/path/a/b/c will
+  // result in "path/a/b/c".
+  string storage_path = 4;
+  // The Artifact URI
+  string uri = 5;
+  // Optional Output. Specifies a signed-url that can be used to
+  // download this Artifact directly from its store.
+  string share_url = 6;
+  // The namespace associated with this Artifact. This is determined
+  // by the namespace of the parent PipelineRun that
+  // the creation of this Artifact.
+  string namespace = 7;
+  // The artifacts type (e.g. system.Dataset)
+  string artifact_type = 8;
+  // The size of the artifact in bytes.
+  // If the artifact does not exist in object store (e.g. Metrics)
+  // then this is omitted.
+  int64 artifact_size = 9;
+  // Creation time of the artifact.
+  google.protobuf.Timestamp created_at = 10;
+  // Last update time of the artifact.
+  google.protobuf.Timestamp last_updated_at = 11;
+
+  // In case any error happens retrieving a artifact field, only artifact ID
+  // and the error message is returned. Client has the flexibility of choosing
+  // how to handle the error. This is especially useful during listing call.
+  google.rpc.Status error = 12;
+}
+
+message GetArtifactRequest {
+  // Required. The ID of the artifact to be retrieved.
+  string artifact_id = 1;
+  // Optional. Set to "true" to included a signed URL with
+  // an expiry (default 15 seconds, unless configured other wise).
+  // This URL can be used to download the Artifact directly from
+  // the Artifact's storage provider. Default to false.
+  string share_url = 2;
+}
+
+// Passed onto MLMD ListOperationOptions
+// https://github.com/kubeflow/pipelines/blob/master/third_party/ml-metadata/ml_metadata/proto/metadata_store.proto#L868
+message ListArtifactRequest {
+  // Optional.
+  // Max number of resources to return in the result. A value of zero or less
+  // will result in the default (20).
+  // The API implementation also enforces an upper-bound of 100, and picks the
+  // minimum between this value and the one specified here.
+  // [default = 20]
+  int32 max_result_size = 1;
+
+  enum Field {
+    FIELD_UNSPECIFIED = 0;
+    CREATE_TIME = 1;
+    LAST_UPDATE_TIME = 2;
+    ID = 3;
+  }
+
+  // Optional. Ordering field. [default = ID]
+  Field order_by_field = 2;
+  // Optional. Order by Ascending. [default = true]
+  bool is_asc = 3;
+
+  // Optional. Identifies the next page of results.
+  string next_page_token = 4;
+
+  // Required. Namespace of the Artifact's context.
+  string namespace = 6;
+}
+
+message ListArtifactResponse {
+  // List of retrieved artifacts.
+  repeated Artifact artifacts = 1;
+
+  // The token to list the next page of artifacts.
+  string next_page_token = 3;
```

Comment:
are you reserving `2` for `total_size` in the future? In that case, should we use `reserved`? ref: https://protobuf.dev/programming-guides/proto3/#reserved-field-names

Reply:
no, this is a mistake, will fix

Reply:
still skipped 2

Reply:
fixed